### PR TITLE
Jcalendar replacement for event date

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Remove jcalendar for event date in advanced search & event search

Before
----------------------------------------
Jcalendar

After
----------------------------------------
Datepicker

Technical Details
----------------------------------------
@seamuslee001 alternate approach to even_end_date  - the same would also be used for relationship active date. The advantage is we keep the submitted field 'intact' rather than wrangling it.

Comments
----------------------------------------
I didn't rename the field - but we could.

### CODE REVIEW

**Comment by seamuslee001** [comment]
This seems to work at least for searching i note there is a change in qill, Tho smart groups haven't been converted right i think. I created 4 separate smart groups. The first 2 were from the standalone find participants page and the last from advanced search. The Form Values are below. 

```
a:25:{s:5:"qfKey";s:36:"4433daaa98e50ed53d621671cfcb407d_208";s:8:"entryURL";s:43:"http://47-test/civicrm/event/search?reset=1";s:9:"sort_name";s:0:"";s:5:"group";a:0:{}s:12:"contact_tags";a:0:{}s:12:"contact_type";a:0:{}s:21:"participant_status_id";a:0:{}s:34:"participant_register_date_relative";s:0:"";s:29:"participant_register_date_low";s:0:"";s:30:"participant_register_date_high";s:0:"";s:8:"event_id";s:0:"";s:13:"event_type_id";s:0:"";s:18:"participant_fee_id";a:0:{}s:14:"event_relative";s:10:"this.month";s:20:"event_start_date_low";s:14:"20191001000000";s:19:"event_end_date_high";s:14:"20191031235959";s:22:"event_date_range_error";s:0:"";s:19:"participant_role_id";a:0:{}s:16:"participant_test";s:1:"0";s:24:"participant_is_pay_later";s:0:"";s:26:"participant_fee_amount_low";s:0:"";s:27:"participant_fee_amount_high";s:0:"";s:8:"custom_4";a:0:{}s:4:"task";s:2:"12";s:8:"radio_ts";s:6:"ts_all";}
```
```
a:25:{s:5:"qfKey";s:37:"4433daaa98e50ed53d621671cfcb407d_7111";s:8:"entryURL";s:43:"http://47-test/civicrm/event/search?reset=1";s:9:"sort_name";s:0:"";s:5:"group";a:0:{}s:12:"contact_tags";a:0:{}s:12:"contact_type";a:0:{}s:21:"participant_status_id";a:0:{}s:34:"participant_register_date_relative";s:0:"";s:29:"participant_register_date_low";s:0:"";s:30:"participant_register_date_high";s:0:"";s:8:"event_id";s:0:"";s:13:"event_type_id";s:0:"";s:18:"participant_fee_id";a:0:{}s:14:"event_relative";s:1:"0";s:20:"event_start_date_low";s:10:"10/01/2019";s:19:"event_end_date_high";s:0:"";s:22:"event_date_range_error";s:0:"";s:19:"participant_role_id";a:0:{}s:16:"participant_test";s:1:"0";s:24:"participant_is_pay_later";s:0:"";s:26:"participant_fee_amount_low";s:0:"";s:27:"participant_fee_amount_high";s:0:"";s:8:"custom_4";a:0:{}s:4:"task";s:2:"12";s:8:"radio_ts";s:6:"ts_all";}
```
```
a:10:{i:0;a:5:{i:0;s:8:"entryURL";i:1;s:1:"=";i:2;s:43:"http://47-test/civicrm/event/search?reset=1";i:3;i:0;i:4;i:0;}i:1;a:5:{i:0;s:21:"group_search_selected";i:1;s:1:"=";i:2;s:5:"group";i:3;i:0;i:4;i:0;}i:2;a:5:{i:0;s:16:"privacy_operator";i:1;s:1:"=";i:2;s:2:"OR";i:3;i:0;i:4;i:0;}i:3;a:5:{i:0;s:14:"privacy_toggle";i:1;s:1:"=";i:2;s:1:"1";i:3;i:0;i:4;i:0;}i:4;a:5:{i:0;s:20:"event_start_date_low";i:1;s:1:"=";i:2;s:14:"20191001000000";i:3;i:0;i:4;i:0;}i:5;a:5:{i:0;s:19:"event_end_date_high";i:1;s:1:"=";i:2;s:14:"20191031235959";i:3;i:0;i:4;i:0;}i:6;a:5:{i:0;s:16:"participant_test";i:1;s:1:"=";i:2;s:1:"0";i:3;i:0;i:4;i:0;}s:14:"relative_dates";a:1:{s:5:"event";s:10:"this.month";}i:7;a:5:{i:0;s:8:"operator";i:1;s:1:"=";i:2;s:3:"AND";i:3;i:0;i:4;i:0;}i:8;a:5:{i:0;s:14:"component_mode";i:1;s:1:"=";i:2;s:1:"1";i:3;i:0;i:4;i:0;}}
```
```
a:8:{i:0;a:5:{i:0;s:8:"entryURL";i:1;s:1:"=";i:2;s:43:"http://47-test/civicrm/event/search?reset=1";i:3;i:0;i:4;i:0;}i:1;a:5:{i:0;s:21:"group_search_selected";i:1;s:1:"=";i:2;s:5:"group";i:3;i:0;i:4;i:0;}i:2;a:5:{i:0;s:16:"privacy_operator";i:1;s:1:"=";i:2;s:2:"OR";i:3;i:0;i:4;i:0;}i:3;a:5:{i:0;s:14:"privacy_toggle";i:1;s:1:"=";i:2;s:1:"1";i:3;i:0;i:4;i:0;}i:4;a:5:{i:0;s:20:"event_start_date_low";i:1;s:1:"=";i:2;s:10:"10/01/2019";i:3;i:0;i:4;i:0;}i:5;a:5:{i:0;s:16:"participant_test";i:1;s:1:"=";i:2;s:1:"0";i:3;i:0;i:4;i:0;}i:6;a:5:{i:0;s:8:"operator";i:1;s:1:"=";i:2;s:3:"AND";i:3;i:0;i:4;i:0;}i:7;a:5:{i:0;s:14:"component_mode";i:1;s:1:"=";i:2;s:1:"1";i:3;i:0;i:4;i:0;}}
```

**Comment by eileenmcnaughton** [comment]
@seamuslee001 did you decide what the issue was with the conversion? Or you didn't dig that far

File: CRM/Upgrade/Incremental/php/FiveTwenty.php [hunk 1]
```diff
@@ -130,6 +130,7 @@ public function upgrade_5_20_alpha1($rev) {
         'mailing_job_start_date',
         'relationship_start_date',
         'relationship_end_date',
+        'event_date',
```
Comment (eileenmcnaughton) [inline CRM/Upgrade/Incremental/php/FiveTwenty.php]: Ah should be just event not event date

**Comment by eileenmcnaughton** [comment]
@seamuslee001 does that fix it?

**Comment by seamuslee001** [comment]
@eileenmcnaughton it seems to me to be threefold 1) the smart groups created by action on the stand alone find participants search doesn’t match the format when done through advanced search 2) the check in is the relative date isn’t in the relative dates array is causing issues with no.4 because the relative = 0 is not being added to the form values and 3) the relative dates upgrade was tricky.

**Comment by eileenmcnaughton** [comment]
@seamuslee001 ug

We just need a special routine for this group? Do you have it clear in your head how it would look?

**Comment by eileenmcnaughton** [comment]
@seamuslee001 I've pulled the conversion bit out of this PR. I figure we are so close now I'll look at the last relationship field & do that & then log_date one & then check the conversion as the last step since it seems it needs to be custom

**Comment by seamuslee001** [comment]
I tested this and confirmed that it works noting we will handle upgrade separately

**Pull request merged**